ipfsapi/client.py

```diff
@@ -2056,7 +2056,12 @@ def shutdown(self):
         ``ipfsapi.exceptions.ConnectionError``, until you start another IPFS
         daemon instance.
         """
-        return self._client.request('/shutdown')
+        try:
+            return self._client.request('/shutdown')
+        except exceptions.ConnectionError:
+            # Sometimes the daemon kills the connection before sending a
+            # response causing an incorrect `ConnectionError` to bubble
```

Comment:
Hmmm... should we file that as a bug with go-ipfs?

Reply:
Indeed! Filed it here: https://github.com/ipfs/go-ipfs/issues/4994